cmd/cluster/etcd_health.go

```diff
@@ -0,0 +1,239 @@
+	"etcdctl endpoint health -w table",
+	"etcdctl member list -w table",
+}
+
+type logCapture struct {
+	buffer bytes.Buffer
+}
+
+func (capture *logCapture) GetStdOut() string {
+	return capture.buffer.String()
+}
+
+func (capture *logCapture) Write(p []byte) (n int, err error) {
+	a := string(p)
+	capture.buffer.WriteString(a)
+	return len(p), nil
+}
+
+func newCmdEtcdHealthCheck(kubeCli client.Client, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	return &cobra.Command{
+		Use:               "etcd-health-check",
+		Short:             "Checks if the etcd component status",
+		Long:              `Checks etcd component health status for member replacement`,
+		Args:              cobra.ExactArgs(0),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(EtcdHealthCheck(kubeCli, flags))
+		},
+	}
+}
+
+func EtcdHealthCheck(kubeCli client.Client, flags *genericclioptions.ConfigFlags) error {
+
+	err := ControlplaneNodeStatus(kubeCli)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	podlist, err := EtcdPodStatus(kubeCli, flags)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	err = EtcdCrStatus(kubeCli)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	var kubeconfig *string
+
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	kconfig, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(kconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	MatchPodName := false
+	var etcdPodName string
+	for !MatchPodName {
+		fmt.Print("Enter Etcd Pod Name: ")
+		fmt.Scan(&etcdPodName)
+		for _, pods := range podlist.Items {
+			if etcdPodName == pods.Name {
+				MatchPodName = true
+			}
+		}
+	}
+
+	for _, v := range etcdctlCmd {
+		output, err := Etcdctlhealth(kconfig, clientset, v, etcdPodName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		fmt.Printf("$ %s\n", v)
+		fmt.Println(output)
+	}
+	return nil
+}
+
+func ControlplaneNodeStatus(kubeCli client.Client) error {
+	nodeList := &corev1.NodeList{}
+	if err := kubeCli.List(context.TODO(), nodeList, client.MatchingLabels{MasterNodeLabel: ""}); err != nil {
+		return err
+	}
+
+	fmt.Println("+----------------------------------------------------------------+")
+	fmt.Printf("|                CONTROLPLANE NODE STATUS                        |\n")
+	fmt.Println("+----------------------------------------------------------------+")
+
+	for _, node := range nodeList.Items {
+		for _, condition := range node.Status.Conditions {
+			if condition.Type == corev1.NodeReady {
+				fmt.Printf("%s\t%s\t%s\n", node.Name, corev1.NodeReady, condition.Status)
```

Comment:
This is just personal opinion, so don't take this as a request to change or anything. But I wonder if, in the interests of output brevity, maybe it could only report the nodes that are in a not-ready state. If everything's all good, it can simply print that as a 1-liner. If there's a node not good, specifically point it out. It might save a little time for an SRE to go and interpret the results.

Reply:
Building on to this - functions that print out information without returning anything can be difficult to test. Can we have this function return a list of nodes similar to EtcdPodStatus below?